cli/js/native_plugins.ts

```diff
@@ -0,0 +1,66 @@
+import { sendSync } from "./dispatch_json.ts";
+import { OP_OPEN_NATIVE_PLUGIN, setPluginAsyncHandler } from "./dispatch.ts";
+import { core } from "./core.ts";
+
+export interface AsyncHandler {
+  (msg: Uint8Array): void;
+}
+
+interface NativePluginOp {
+  dispatch(
+    control: Uint8Array,
+    zeroCopy?: ArrayBufferView | null
+  ): Uint8Array | null;
+  setAsyncHandler(handler: AsyncHandler): void;
+}
+
+class NativePluginOpImpl implements NativePluginOp {
+  constructor(private readonly opId: number) {}
+
+  dispatch(
+    control: Uint8Array,
+    zeroCopy?: ArrayBufferView | null
+  ): Uint8Array | null {
+    return core.dispatch(this.opId, control, zeroCopy);
+  }
+
+  setAsyncHandler(handler: AsyncHandler): void {
+    setPluginAsyncHandler(this.opId, handler);
+  }
+}
+
+// TODO(afinch7): add close method.
```

Comment:
I'd be cool if you could implement `close` method - although I'm pretty sure 95% use-cases will be open plugin and forget it still be good to have cleanup method

Reply:
Maybe add this in a follow up PR.